### PR TITLE
Changes for audit logging

### PR DESCRIPTION
**Pull request by krishnangovindraj**
## What is the goal of this PR?
Refactors configuration & grpc service classes for audit logging.

## What are the changes implemented in this PR?
* Refactors grpc service implementation classes, introducing methods which subtypes can override to log events
* Refactors configuration `CoreConfigParser.Log.Output.File` to have filename awareness, allowing it to be re-used for audit log outputs configuration.

### CODE REVIEW

File: dependencies/vaticle/repositories.bzl [hunk 1]
```diff
@@ -20,8 +20,8 @@ load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
 def vaticle_dependencies():
     git_repository(
         name = "vaticle_dependencies",
-        remote = "https://github.com/vaticle/dependencies",
-        commit = "dc35e1d26515102c58e6d44bc23632d953fcc722",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
+        remote = "https://github.com/krishnangovindraj/dependencies",
+        commit = "b910cbfc2a5270439f2ce840150d2020354918d1",  # sync-marker: do not remove this comment, this is used for sync-dependencies by @vaticle_dependencies
```
Comment (krishnangovindraj) [inline dependencies/vaticle/repositories.bzl:24]: TODO

File: server/TransactionService.java [hunk 2]
```diff
@@ -226,17 +226,41 @@ private static TypeDB.Transaction transaction(SessionService sessionSvc, Transac
...
+    }
+
+    protected void executeTypeRequest(TransactionProto.Transaction.Req req) {
+        services.type.execute(req);
+    }
+
+    protected void executeRuleRequest(TransactionProto.Transaction.Req req) {
+        services.rule.execute(req);
+    }
+
```
Comment (krishnangovindraj) [inline server/TransactionService.java:263]: New methods for enterprise to override & add audit logging to.

File: server/parameters/CoreConfig.java [hunk 3]
```diff
@@ -204,6 +209,13 @@ public static class File extends Type {
                     public Path baseDirectory() {
                         return baseDirectory;
                     }
+                    public String filename() {
+                        return filename;
+                    }
+
+                    public String extension() {
+                        return extension;
+                    }
```
Comment (krishnangovindraj) [inline server/parameters/CoreConfig.java:218]: Filename awareness allows it to be re-used as audit log outputs.
The question is whether this class should be moved to the top level of the config (`CoreConfig.Output`) or remain at `CoreConfig.Log.Output`

File: server/parameters/CoreConfig.java [hunk 4]
```diff
@@ -191,10 +191,15 @@ public static class File extends Type {
                     private final YAMLParser.Value.TimePeriodName archiveGrouping;
                     private final YAMLParser.Value.TimePeriod archiveAgeLimit;
                     private final long archivesSizeLimit;
+                    private final String filename;
+                    private final String extension;
 
-                    File(Path baseDirectory, long fileSizeLimit, YAMLParser.Value.TimePeriodName archiveGrouping,
+                    File(Path baseDirectory, String filename, String extension,
```
Comment (flyingsilverfin) [inline server/parameters/CoreConfig.java:197]: So the config output has two new paramters, `filename` and `extension`?
Does this mean we've changed the config yaml file?

File: server/parameters/CoreConfigParser.java [hunk 5]
```diff
@@ -208,7 +210,7 @@ protected static class Type extends Compound<CoreConfig.Log.Output.Type> {
                         "type", "Type of output to define.", restricted(STRING, list(Stdout.type, File.type))
                 );
                 protected static final Compound<CoreConfig.Log.Output.Type.Stdout> stdout = new Stdout();
-                protected static final Compound<CoreConfig.Log.Output.Type.File> file = new File();
+                protected static final Compound<CoreConfig.Log.Output.Type.File> file = new File(TYPEDB_LOG_FILE_NAME, TYPEDB_LOG_FILE_EXT);
```
Comment (krishnangovindraj) [inline server/parameters/CoreConfigParser.java:213]: Marker for other discussion - We set the filename and extension here.

File: server/parameters/CoreConfig.java (hunk 3 above)
Comment (flyingsilverfin) [reply server/parameters/CoreConfig.java:219]: I dont think we should be reusing the classes across different places (Log/Audit). This is my main discomfort right now and i'm not sure how to resolve it...

Comment (flyingsilverfin) [reply server/parameters/CoreConfig.java:219]: let's move the output definitions to a `CoreConfig.Common` which will hold structures used by multiple other parts of the definitions?

**Review by flyingsilverfin — APPROVED** [review]

**Pull request merged**